### PR TITLE
Hibernate panache codestart

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: docs/src/main/asciidoc/hibernate-orm-panache-kotlin.adoc [hunk 1]
```diff
@@ -24,7 +24,7 @@ change or two.  To Panache-enable your entity, you would define it something lik
 [source,kotlin]
 ----
 @Entity
-class Person: PanacheEntity {
```
Comment (ia3andy) [inline docs/src/main/asciidoc/hibernate-orm-panache-kotlin.adoc:27]: Are u sure this is needed in kotlin?

Is it not possible to use a `data` class?

Comment (loicmathieu) [reply docs/src/main/asciidoc/hibernate-orm-panache-kotlin.adoc:27]: This is just a fix of the documentation that misses the parenthesis.

Whether or not data classes can be used with Hibernate is a question for @evanchooly and I guess, as he is the author of the guide, that they are not a good fit for it.

**Comment by loicmathieu** [comment]
@ia3andy I merged the codestarts

**Comment by Sanne** [comment]
Is there any particular reason for the non-panache entity to use getters+setters while the Panache version uses filed instrumentation?

Unless I'm missing something, it would be nice to showcase that both approaches can be used without the "ceremonial" getters and setters overhead.

**Comment by loicmathieu** [comment]
@Sanne I was thinking that the field enhancement was a Panache thing, if you look at the Hibernate ORM guide it shows a regular JPA entity: https://quarkus.io/guides/hibernate-orm#setting-up-and-configuring-hibernate-orm

I was thinking the bytecode transformer to transform field access to accesor was this one: https://github.com/quarkusio/quarkus/blob/main/extensions/panache/panache-common/deployment/src/main/java/io/quarkus/panache/common/deployment/PanacheFieldAccessMethodVisitor.java

But maybe I'm missing something ?

**Comment by Sanne** [comment]
@loicmathieu no it's an Hibernate ORM thing, Panache only adds additional capabilities to it

**Comment by loicmathieu** [comment]
@Sanne it should be documented then :), as on the Panache guide it is said that it's a Panache feature (see point 2): https://quarkus.io/guides/hibernate-orm-panache#how-and-why-we-simplify-hibernate-orm-mappings

I'll update the quickstart.

**Comment by Sanne** [comment]
> @Sanne it should be documented then :), as on the Panache guide it is said that it's a Panache feature (see point 2): https://quarkus.io/guides/hibernate-orm-panache#how-and-why-we-simplify-hibernate-orm-mappings
> 
> I'll update the quickstart.

ah good point - yes I see what you mean. To be clear the Panache doc isn't entirely wrong, as when using Panache the getters/setters are added by Panache - but the doc is indeed misleading as ORM "alone" would do the same.

**Comment by loicmathieu** [comment]
@Sanne I updated the Hibernate ORM with public fields instead of getters§setters

**Review by gsmet — DISMISSED** [review]
This shouldn't be merged until we figure out https://github.com/quarkusio/quarkus/issues/22773 .

**Comment by yrodiere** [comment]
I closed #22773 as not planned. @loicmathieu can you please rebase your PR? Thanks.

**Comment by FroMage** [comment]
> ah good point - yes I see what you mean. To be clear the Panache doc isn't entirely wrong, as when using Panache the getters/setters are added by Panache - but the doc is indeed misleading as ORM "alone" would do the same.

@Sanne is it _really_ the case? I remember from years ago that the bytecode enhancer for field accesses wasn't enabled for Hibernate without Panache. The getters are generated alright, but nothing is there in Quarkus to rewrite the field accesses from outside the entity. Only Panache does that AFAIR.

**Comment by Sanne** [comment]
@FroMage it's called extended enhancement in Hibernate ORM, it's disabled by default but you could enable it to let it do it for you.

https://github.com/hibernate/hibernate-orm/blob/674b1dfd271a940b91038a7e5f72a3f381697e75/hibernate-core/src/main/java/org/hibernate/bytecode/enhance/spi/EnhancementContext.java#L81-L89

**Comment by yrodiere** [comment]
> @Sanne so public fields rewritting is enabled by default for Quarkus or not ?

It is. I worked on that.

File: devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/hibernate-orm-codestart/java/src/main/java/org/acme/MyEntity.tpl.qute.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    @GeneratedValue
+    public Long id;
+
+    public String field;
+}
+{/if}
+{#if input.selected-extensions-ga.contains('io.quarkus:quarkus-hibernate-orm-panache')}
+        import io.quarkus.hibernate.orm.panache.PanacheEntity;
+
+        import javax.persistence.Entity;
```
Comment (ia3andy) [inline devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/hibernate-orm-codestart/java/src/main/java/org/acme/MyEntity.tpl.qute.java:38]:
```suggestion
import io.quarkus.hibernate.orm.panache.PanacheEntity;
import javax.persistence.Entity;
```

**Review by ia3andy — DISMISSED** [review]
Looking good

**Comment by loicmathieu** [comment]
@yrodiere PR is updated and all test pass, can you approve it ?

**Comment by yrodiere** [comment]
@loicmathieu sorry, I was on leave. Will have a look ASAP.

**Review by yrodiere — APPROVED** [review]
I wasn't able to test this locally (can't seem to use codestarts from a locally-built Quarkus). But the code looks good, so I'll just trust the tests :)

**Pull request merged**